docs/docker.rst

```diff
@@ -0,0 +1,160 @@
+
+- ``newgrp docker`` or ``su - $USER``
+
+- ``sudo usermod -a -G docker $USER``
+
+Get `docker-compose
+<https://docs.docker.com/compose/>`_:
+
+.. code-block:: console
+
+ sudo pip install docker-compose
+
+- Add ``DOCKER_DATA`` env variable in your ``.bashrc`` or ``.zshrc``. In this directory you will have all the persistent data between Docker runs.
+
+.. code-block:: console
+
+ shell mkdir -p "$DOCKER_DATA"
+ export DOCKER_DATA=~/inspirehep_docker_data/
+
+By default the virtualenv and everything else will be kept on ``/tmp`` and they will be available only until the next reboot.
+
+- Install a host persistent venv and build assets
+
+.. code-block:: console
+
+ docker-compose pull
+ docker-compose -f docker-compose.deps.yml run --rm pip
+
+.. note:: If you have trouble with internet connection inside docker probably you are facing known
+          DNS issue. Please follow `this solution
+          <http://askubuntu.com/questions/475764/docker-io-dns-doesnt-work-its-trying-to-use-8-8-8-8/790778#790778>`_
+          with DNS: ``--dns 137.138.17.5 --dns 137.138.16.5``.
+
+.. code-block:: console
+
+ docker-compose -f docker-compose.deps.yml run --rm assets
+
+- Run the service locally
+
+.. code-block:: console
+
+ docker-compose up
+
+Go to ``localhost:5000``
+
+- Populate database
+
+.. code-block:: console
+
+ docker-compose run --rm web scripts/recreate_records
+
+- Run tests in an **isolated** environment.
+
+  You can choose one of the following tests types:
+
+  - unit
+  - disambiguation
+  - workflows
+  - integration
+  - acceptance
+
+.. code-block:: console
+
+ docker-compose -f docker-compose.test.yml run --rm <tests type>
+ docker-compose -f docker-compose.test.yml down
+
+.. tip:: - cleanup all the containers:
+
+           ``docker rm $(docker ps -qa)``
+
+         - cleanup all the images:
+
+           ``docker rmi $(docker images -q)``
+
+         - cleanup the virtualenv (careful, if docker_data is set to something you care about, it will be removed):
+
+           ``sudo rm -rf "${DOCKER_DATA?DOCKER_DATA was not set, ignoring}"``
+
+Extra useful tips
+#################
+
+- Run a random shell
+
+After ``docker-compose up`` just run:
+
+.. code-block:: console
+
+ docker-compose run --rm web inspirehep shell
+
+- Run *virtualenv* bash shell for running scripts manually (e.g. recreating records or `building documentation`_)
+
+.. _building documentation: http://inspirehep.readthedocs.io/en/latest/building_the_docs.html
+
+With ``docker-compose up`` just run:
+
+.. code-block:: console
+
+ docker-compose run --rm web bash
+
+- Reload code in a worker
+
+With ``docker-compose up`` just run:
+
+.. code-block:: console
+
+ docker-compose restart worker
+
+- Quick and safe reindex
+
+With ``docker-compose up`` just run:
+
+.. code-block:: console
+
+ docker-compose restart worker && docker-compose run --rm web scripts/recreate_records
```

Comment:
You can add here:

Recreate all static assets. Will download all dependencies from npm and copy all static files to `${DOCKER_DATA}/tmp/virtualenv/var/inspirehep-instance/static`

`docker-compose -f docker-compose.deps.yml run --rm assets`